icu4c/source/test/intltest/uts46test.cpp

```diff
@@ -160,7 +160,7 @@ void UTS46Test::TestAPI() {
     char buffer[100];
     TestCheckedArrayByteSink sink(buffer, UPRV_LENGTHOF(buffer));
     errorCode=U_ZERO_ERROR;
-    nontrans->labelToUnicodeUTF8(StringPiece(NULL, 5), sink, info, errorCode);
+    nontrans->labelToUnicodeUTF8(StringPiece((const char *)NULL, 5), sink, info, errorCode);
```

Comment:
I think this cast should be unnecessary.

Reply:
I need the cast for NULL. I didn't add a nullptr_t overload for this constructor because I don't want to promote the use of nullptr with an explicit length, but this particular unit test case is for coverage of handling of bad input.